routes/models/proposal.js

```diff
@@ -0,0 +1,132 @@
+
+    var total_qry = "select count(*) as count from "+DB.proposals ;
+
+    DB.conn.queryAsync(total_qry).then(function(count) {
+        if(count[0].count == 0){
+          res({
+            "statusCode":200, "message":"Proposals listing done","data":{
+              "count":0,
+              "proposals":[]
+            }
+          });    
+        }
+        else {
+          DB.conn.queryAsync(options).then(function(proposals) {
+            res({"statusCode":200, "message":"Proposals listing done","data":{
+              "count": count[0].count ,
+              "proposals":proposals
+            }});
+          });
+        }        
+    });
+}
+
+module.exports.update = function(req, res) {
+    DB.conn.queryAsync('SELECT * FROM '+DB.proposals+' where id = '+req.payload.id+'').then(function(rows) {
+        if(rows.length == 0) {
+          res(Boom.notFound('No proposal found with this id . Please enter the correct proposal ID'));
+        }
+        else {
+          var dt =  req.payload ;
+          DB.conn.queryAsync('UPDATE '+DB.proposals+' SET title = ? , more_info = ?, client_id = ?,assigned_to = ? where id = ?',[dt.title, dt.more_info, dt.client_id, dt.assigned_to,dt.id]).then(function(result) {
+              res({"statusCode":200,"message":"Proposal updated successfully!"});
+          });
+        }
+    });
+}
+
+module.exports.view = function(req,res) {    
+    var qry = 'SELECT P.id, P.title, P.more_info , P.created ,  U.id, U.first_name , C.id , C.name FROM '+DB.proposals+' P INNER JOIN '+DB.users+' U on P.created_by = U.id INNER JOIN '+DB.clients+' C on P.client_id = C.id where P.id = '+req.params.Id ; 
+    var options = {sql: qry , nestTables: true } ;
+    DB.conn.queryAsync(options).then(function(proposal) {
+        if(proposal.length) {
+            var pr_qry = 'SELECT S.* FROM '+DB.specifications +' S where S.proposal_id = '+req.params.Id ;
+            DB.conn.queryAsync(pr_qry).then(function(specifications) {
+              res({"statusCode":200, "message":"Proposal details", "data":{"proposal":proposal[0],"specifications": specifications}});
+            });
+        }
+        else
+        {
+            res(Boom.notFound('No proposal found with this id . Please enter the correct proposal ID'));
+        }
+    });
+}
+
+module.exports.assign = function(req,res) {
+    var assigned_to  =  req.payload.assigned_to;
+    var proposal_id  =  req.payload.proposal_id;
+    var qry = "UPDATE "+ DB.proposals+" set assigned_to = ? where id = ?" ;
+    DB.conn.queryAsync(qry,[assigned_to, proposal_id]).then(function(response) {
+        res({"statusCode":200, "message":"Proposal assigned to user"}) ;
+    });
+}
+
+module.exports.change_client = function(req,res) {
+    var client_id = req.payload.client_id;
+    var proposal_id = req.payload.proposal_id;
+    var qry = "UPDATE "+ DB.proposals+" set client_id = ? where id = ?" ;
+    DB.conn.queryAsync(qry,[client_id, proposal_id]).then(function(response) {
```

Comment:
@tusharncuelogic @sandeepp-cuelogic @gauravc-cuelogic @pritig-cuelogic exception handling should be there.